cirq-google/cirq_google/engine/engine_job.py

```diff
@@ -302,6 +310,17 @@ async def results_async(self) -> Sequence[EngineResult]:
         return self._results
 
     async def _await_result_async(self) -> quantum.QuantumResult:
+        if self._stream_job_response_future is not None:
+            response = await self._stream_job_response_future
+            if isinstance(response, quantum.QuantumResult):
+                return response
+            elif isinstance(response, quantum.QuantumJob):
+                self._job = response
+                _raise_on_failure(response)
+            else:
+                # coverage: ignore
+                raise ValueError('Internal error: The stream response type is not recognized.')
+
         async with duet.timeout_scope(self.context.timeout):  # type: ignore[arg-type]
```

Comment:
If the response future is `None`, could/should we make a `GetResultRequest` over the stream rather than polling?

Reply:
Once streaming becomes the only mode of operation, we could remove this branch of logic.

Reply:
Just to be clear - I _think_ that the reason that you're citing is that we can't disable streaming with the feature flag if we migrate the legacy code.